src/solver/naive_solver.rs

```diff
@@ -0,0 +1,208 @@
+    {
+        if best_sell_order_token0.have_price_overlap(&best_sell_order_token1) {
+            return Solution {
+                sell_orders_token0: vec![best_sell_order_token0.clone()],
+                sell_orders_token1: vec![best_sell_order_token1.clone()],
+            };
+        };
+    };
+
+    Solution {
+        sell_orders_token0: vec![],
+        sell_orders_token1: vec![],
+    }
+}
+
+fn check_orders_sorted_by_limit_price(orders: &Vec<Order>) -> bool {
+    for (i, x) in orders.iter().enumerate() {
+        for y in orders.iter().skip(i + 1) {
+            if x > y {
+                return false;
+            }
+        }
+    }
+    true
+}
+
+#[cfg(test)]
+pub mod test_util {
+    use super::*;
+    use ethcontract::web3::types::{Address, H256};
+
+    // Define some constants for test orders.
+    const TOKEN0_ADDRESS: &'static str = "A193E42526F1FEA8C99AF609dcEabf30C1c29fAA";
+    const TOKEN1_ADDRESS: &'static str = "FDFEF9D10d929cB3905C71400ce6be1990EA0F34";
+    const OWNER: &'static str = "63FC2aD3d021a4D7e64323529a55a9442C444dA0";
+
+    const SIGNATURE_R: &'static str =
+        "07cf23fa6f588cc3a91de8444b589e5afbf91c5d486c512a353d45d02fa58700";
+    const SIGNATURE_S: &'static str =
+        "53671e75b62b5bd64f91c80430aafb002040c35d1fcf25d0dc55d978946d5c11";
+
+    fn generate_standard_order_with_defined_sell_and_buy_amount(
+        sell_amount: &str,
+        buy_amount: &str,
+    ) -> Order {
+        Order {
+            sell_amount: U256::from_dec_str(sell_amount).unwrap(),
+            buy_amount: U256::from_dec_str(buy_amount).unwrap(),
+            current_sell_amount: None,
+            current_buy_amount: None,
+            sell_token: address(TOKEN0_ADDRESS),
+            buy_token: address(TOKEN1_ADDRESS),
+            owner: address(OWNER),
+            nonce: 0,
+            signature_v: 27 as u8,
+            signature_r: signature(SIGNATURE_R),
+            signature_s: signature(SIGNATURE_S),
+            valid_until: U256::from("0"),
+        }
+    }
+    fn generate_standard_opposite_order_with_defined_sell_and_buy_amount(
+        sell_amount: &str,
+        buy_amount: &str,
+    ) -> Order {
+        Order {
+            sell_amount: U256::from_dec_str(sell_amount).unwrap(),
+            buy_amount: U256::from_dec_str(buy_amount).unwrap(),
+            current_sell_amount: None,
+            current_buy_amount: None,
+            sell_token: address(TOKEN1_ADDRESS),
+            buy_token: address(TOKEN0_ADDRESS),
+            owner: address(OWNER),
+            nonce: 0,
+            signature_v: 27 as u8,
+            signature_r: signature(SIGNATURE_R),
+            signature_s: signature(SIGNATURE_S),
+            valid_until: U256::from("0"),
+        }
+    }
+    fn address(address_str: &str) -> Address {
+        return (address_str).parse().unwrap();
+    }
+
+    fn signature(signature_str: &str) -> H256 {
+        return (signature_str).parse().unwrap();
+    }
```

Comment:
Maybe these could go in a top-level `test` module that only gets included for tests and can be shared across the project. It can even contain a "prelude" for commonly used types during tests:

- `lib.rs`
```rust
#[cfg(test)]
mod test;
```

- `test.rs`
```rust
pub fn address(address_str: &str) -> Address {
    return (address_str).parse().unwrap();
}

pub fn signature(signature_str: &str) -> H256 {
    return (signature_str).parse().unwrap();
}

pub mod prelude {
    pub use super::*;
    pub use ethcontract::{Address, H256};
}
```

Then for unit tests:
```rust
#[cfg(test)]
mod tests {
    use crate::test::prelude::*;

    #[test]
    fn some_test() {
        // ...
    }
}
```